src/core/compact_object.h

```diff
@@ -325,6 +325,9 @@ class CompactObj {
   void SetExternal(size_t offset, size_t sz);
   std::pair<size_t, size_t> GetExternalSlice() const;
 
+  // The opposite of SetExternal, changes the external entry to be an in-memory string.
+  void Materialize(std::string_view str);
```

Comment:
Maybe call this `SetExternalString()`?
This is pretty much what it does, right? `SetString()` but forces it to be non-inline
(I'm not exactly sure why you named it Materialize(), perhaps there's a larger context)

Reply:
Sorry, I guess it doesn't force, but rather `CHECK()`s...
I'm not sure why it's needed over just calling `SetString()` but I'm sure you've got your reasons :rofl: